### ISSUE TITLE
Loss nan in MaskRCNN

### ISSUE THREAD

**Issue by DeVyacheslav** [learnml-today/object-detection-with-pytorch#25]
Hi, I started MaskRCNN training on nuclei detection data science bowl 2018 dataset and was faced with an issue in model Loss: first 5 to 8 batches (size=3) could be trained with Loss = ~2.5, but after that Loss becomes nan. I checked each loss from ['loss_objectness','loss_rpn_box_reg', 'loss_classifier','loss_box_reg','loss_mask'], and I found the problem in loss_mask. The issue was in mask normalization, they should be [0, 1] while in dataset that was [0, 255]. After this fix the Loss nan issue appeared again, but this time the problem was in loss_box_reg part. I checked the boxes and everything seems fine like in the same way as in COCO dataset.
Any suggestions?

**Comment by creotiv** [comment]
it's hard without seeing the code and metrics. 
The problem can appear in:
1) problem with data input. many people doing mistakes in data generators which cause problems with training (like normalization that you said.). I recommend always write visual tests on generators, to be sure that they are working like expected.
2) networks that consist of different parts and many losses can have problems with loss normalization, and in the end, this results in gradient explode or vanish. You can fix this with different gradient normalization techniques like clipping, or just try to minimize loss. By my experience, MaskRcnn can lead to this problems with LR < 1e-3 (but this related to the data).
3) check places which you changed in base network(if this was done)

Also i recommend you to look at this thread: https://www.kaggle.com/c/data-science-bowl-2018/discussion/56326?fbclid=IwAR23S_ASnzsiFTOw_itoD0wD2z36aNBfxfS0Q3kgkwaxF_5IWxApvCOaghY#593204

Also you can share your colab here, so i can run it and debug to see where is the problem can be(but this will take some time)
